src/main/java/org/opensearch/knn/index/query/ExactSearcher.java

```diff
@@ -59,23 +60,27 @@ public class ExactSearcher {
      */
     public Map<Integer, Float> searchLeaf(final LeafReaderContext leafReaderContext, final ExactSearcherContext exactSearcherContext)
         throws IOException {
-        KNNIterator iterator = getKNNIterator(leafReaderContext, exactSearcherContext);
+        final Optional<KNNIterator> iterator = getKNNIterator(leafReaderContext, exactSearcherContext);
```

Comment:
I mentioned it in the description why I am preferring optional over null. The idea was to enforce the null/empty checks otherwise this bug can come in code again. So if you have a better suggestion to enforce this I am happy to change. But otherwise I would like to keep optional to prefer correctness here.

Reply:
The only I can think of to avoid Optional and enforce a null check would be not to return nulls and throw compile time exceptions. This will enforce the user of the code to catch the exception in case of field not present and handle it properly.

Reply:
@shatejas , @Vikasht34 , @jmazanec15 please let me know your thoughts. If we are agreeing to have a null returns then its also fine. But in that case we will open ourselves for NPE.

Reply:
Another option is throwing an empty iterator.

Reply:
@heemin32 I thought about it but since there is no point in moving ahead in the search if the segment contain no vector field, we would have have the checks for empty knnIterator class, which I am trying to see how it will be different than optional here. Also, if we don't exit early then other attributes like space type, etc we need to handle the nulls for FieldInfo. Hence I was thinking optional is a better approach to enforce null checks. 

Other option can be we create a function to get the field info and have good java docs around it to check for null, but again its a best effort.  

So, in a way I see Optional will just work here. Please let me know your thoughts. Open for suggestions.

Reply:
> I mentioned it in the description why I am preferring optional over null. The idea was to enforce the null/empty checks otherwise this bug can come in code again. So if you have a better suggestion to enforce this I am happy to change. But otherwise I would like to keep optional to prefer correctness here.

I will leave upto you to come up with better approach which we can avoid optional  still he solves the purpose , before Java 8 , where optional was not introduced, here are the traditional way to solve this 
1. Using Traditional Null Checks
2.Returning a Default or No-Op Implementation or the one @heemin32 suggested.
3.Throwing an Exception for Missing Cases.


https://homes.cs.washington.edu/~mernst/advice/nothing-is-better-than-optional.html


``` 
In conclusion, using regular Java references is better than special libraries such as Optional.

    Every error that you might commit when working with nullable references has a direct analogue that you might commit when working with Optional, and the consequences (a run-time exception) are equally as severe. Thus, Optional alone doesn't solve any problems.
    Optional creates new potential problems.
    Optional's syntax is uglier than use of nullable references.
    Optional is less efficient than use of nullable references.
    The motivation of Optional is to never forget to do a check. Although Optional does not guarantee this, other tools do. Those tools, such as the [Nullness Checker](https://checkerframework.org/manual/#nullness-checker), are more powerful and precise, and they give a compile-time guarantee.
```

Reply:
This has come up multiple times, I think we should settle on whether to use Optional based on data. The plugin is memory sensitive and we should see how much of an effect it has in the overall performance if its used just as a optional as well as with optional chaining.

@navneet1v Looking at the usage in the PR, I don't see major benefit of using Optional in private methods as its not reused and the method is for used for single-responsibility. On the other hand, in public methods it is extremely useful to indicate it can be null, forcing the client to handle it. 

Overall there are pros to using optional monoid :
1. Clear indication for client of the API to indicate 
2. Chaining with Optional reduces verbosity, increases maintainability
3. Easier testing since you can avoid branching

All these benefits are moot if its affects performance in any way. As far as this PR is concerned I don't foresee a major performance impact so I am okay with it

Reply:
Ideally, `FieldInfo` should be made optional rather than `KNNIterator`. The iterator doesn’t need to be null since we can return an empty iterator instead.
By the way, I’m okay with either the current approach or using null.

Reply:
Resolving this comment as I have removed the optional from the code and moved getting the fieldInfo to a utility function.